src/Function.cs

```diff
@@ -59,6 +68,51 @@ public static Function FromCallback(IStore store, Delegate callback)
                 return new Function(store, externFunc, parameterKinds, resultKinds);
             }
         }
+
+        /// <summary>
+        /// Creates an function given an untyped callback.
+        /// </summary>
+        /// <param name="store">The store to create the function in.</param>
+        /// <param name="callback">The callback for when the function is invoked.</param>
+        /// <param name="parameters">The function parameter kinds.</param>
+        /// <param name="results">The function result kinds.</param>
+        public static Function FromCallback(IStore store, UntypedCallbackDelegate callback, IReadOnlyList<ValueKind> parameters, IReadOnlyList<ValueKind> results)
+        {
+            if (store is null)
+            {
+                throw new ArgumentNullException(nameof(store));
+            }
+
+            if (callback is null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            // Copy the lists to ensure they are not externally modified.
+            var parameterKinds = new List<ValueKind>(parameters);
+            var resultKinds = new List<ValueKind>(results);
+
+            using var funcType = GetFunctionType(parameterKinds, resultKinds);
+
+            unsafe
+            {
+                Native.WasmtimeFuncCallback func = (env, callerPtr, args, nargs, results, nresults) =>
+                {
+                    return InvokeUntypedCallback(callback, callerPtr, args, (int)nargs, results, (int)nresults, resultKinds);
+                };
+
+                Native.wasmtime_func_new(
+                    store.Context.handle,
+                    funcType,
+                    func,
+                    GCHandle.ToIntPtr(GCHandle.Alloc(func)),
+                    Finalizer,
+                    out var externFunc
+                );
+
+                return new Function(store, externFunc, parameterKinds, resultKinds);
```

Comment:
Do we need a `GC.KeepAlive` above this?

Reply:
You're right, this is needed here (I just copied the method here from the above one, and because #206 wasn't merged yet, I forgot to add the call here 😆)